src/atomate2/abinit/jobs/response.py

```diff
@@ -0,0 +1,430 @@
+        likely the SCF performed to get the WFK.
+    use_symmetries : True if only the irreducible perturbations should
+        be returned, False otherwise.
+    qpt: qpoint of the phonon in reduced coordinates. Used to shift the k-mesh
+        if qpt is not passed, gsinput must already contain "qpt"
+        otherwise an exception is raised.
+    """
+    if use_symmetries:
+        perts = gsinput.abiget_irred_phperts(qpt=qpt)  # TODO: quid manager?
+    else:
+        perts = [{"idir": 1}, {"idir": 2}, {"idir": 3}]
+
+    return perts
+
+
+@job
+def generate_dde_perts(
+    gsinput: AbinitInput,
+    # TODO: or gsinput via prev_outputs?
+    use_symmetries: bool | None = False,
+) -> list[dict[str, int]]:
+    """
+    Generate the perturbations for the DDE calculations.
+
+    Parameters
+    ----------
+    gsinput : an |AbinitInput| representing a ground state calculation,
+        likely the SCF performed to get the WFK.
+    use_symmetries : True if only the irreducible perturbations should
+        be returned, False otherwise.
+    """
+    if use_symmetries:
+        perts = gsinput.abiget_irred_ddeperts()  # TODO: quid manager?
+    else:
+        perts = [{"idir": 1}, {"idir": 2}, {"idir": 3}]
+
+    return perts
+
+
+@job
+def generate_dte_perts(
+    gsinput: AbinitInput,
+    # TODO: or gsinput via prev_outputs?
+    skip_permutations: bool | None = False,
+    phonon_pert: bool | None = False,
+    ixc: int | None = None,
+) -> list[dict[str, int]]:
+    """
+    Generate the perturbations for the DTE calculations.
+
+    Parameters
+    ----------
+    gsinput : an |AbinitInput| representing a ground state calculation,
+        likely the SCF performed to get the WFK.
+    skip_permutations: Since the current version of abinit always performs
+        all the permutations of the perturbations, even if only one is asked,
+        if True avoids the creation of inputs that will produce duplicated outputs.
+    phonon_pert: is True also the phonon perturbations will be considered.
+        Default False.
+    ixc: Value of ixc variable. Used to overwrite the default value read
+        from pseudos.
+    """
+    # Call Abinit to get the list of irreducible perturbations
+    gsinput = gsinput.deepcopy()
+    gsinput.pop_vars(["autoparal", "npfft"])
+    perts = gsinput.abiget_irred_dteperts(
+        phonon_pert=phonon_pert, ixc=ixc
+    )  # TODO: quid manager?
+
+    if skip_permutations:
+        perts_to_skip: list = []
+        reduced_perts = []
+        for pert in perts:
+            p = (
+                (pert.i1pert, pert.i1dir),
+                (pert.i2pert, pert.i2dir),
+                (pert.i3pert, pert.i3dir),
+            )
+            if p not in perts_to_skip:
+                reduced_perts.append(pert)
+                perts_to_skip.extend(itertools.permutations(p))
+
+        perts = reduced_perts
+
+    return perts
+
+
+@job
+def run_ddk_rf(
```

Comment:
you mean a common run_response right?

Reply:
not necessarily a single job function. I was thinking more to a function that contains most of the code that is shared and is called by the jobs. But if you find that it would be easier to have a single Job that is also an option.